### DISCUSSION TITLE
A visual editor for Data Flows

### DISCUSSION THREAD

**Discussion by loleg** [frictionlessdata/frictionlessdata.io#607] [Ideas]
Node-based visual editors are popular with creative tools ([Blender](https://docs.blender.org/manual/en/latest/editors/texture_node/introduction.html), [PraxisLIVE](https://github.com/praxis-live), [Buzz Tracker](http://jeskola.net/buzz/)), and they make sense for data wrangling environments too. Let's support the "no code" movement, and make it possible to use [Data Flows](https://github.com/datahq/dataflows) within a graphical, explorative environment. [Node RED](https://nodered.org/) is a tool that is popular in the open hardware community, and we could even as a first step try to make a prototype with it.

Note: something @n0rdlicht and I mentioned in our [presentation](https://hackmd.io/@oleg/Hk9i_rAaI#/13) in the FD meetup.

**Comment by aborruso** [comment]
I'm adding another tool to chekc, the great https://n8n.io/

![image](https://user-images.githubusercontent.com/30607/85836771-68896180-b797-11ea-939c-1876c16d51c2.png)

**Comment by loleg** [comment]
As an inspiring example of this, I've been working with [Orchest](https://github.com/orchest/orchest) a new, AGPL open source tool for data science workflows, which in its design reflects our principles of simple, filesystem and web-friendly structures, and project pipelines that echo School of Data workshops. Adding Frictionless Data support to it would be an interesting way to connect the dots. 

> "The mental model in Orchest is centered around Projects. Within each project you get to create multiple pipelines through the Orchest UI, and every pipeline consists of pipeline steps that point to your scripts."

![](https://orchest.readthedocs.io/en/latest/_images/run-any-subset.png)

See https://orchest.readthedocs.io/

**Comment by augusto-herrmann** [comment]
Good idea, @loleg !

[Frictionless Transform Pipelines](https://framework.frictionlessdata.io/docs/guides/transform-guide/#transforming-pipeline) already use JSON as a possible way to describe data transformation. Perhaps data transformation pipelines should also be part of the [Frictionless specification](https://specs.frictionlessdata.io/) so that a visual editor could be used with any tool, e.g. both with [Frictionless Transform](https://github.com/frictionlessdata/frictionless-py/) and with Data Flows?

What do you think, @lwinfree @roll ?
